### DISCUSSION TITLE
can I have one build script for multiple packages?

### DISCUSSION THREAD

**Discussion by csenio** [vercel/turborepo#559] [Help] [answered]
Hey everyone, 
I'm new to monorepos and was wondering if I could have one build config at the root of my project that builds each of my packages.

My project structure looks like this
```
|-packages
    |-button
        |-package.json
        |-src
           |-index.js
           |-button.svelte
    |-select
        |-package.json
        |-src
           |-index.js
           |-select.svelte
etc...
```

Previously, each of these packages had its own `rollup.config.js` and `build` script. I would prefer to have a single `rollup.config.js` and `build` script somewhere in the root of the project (that still gives each package its own `dist` folder). Is that possible with turborepo?

**Comment by qdozaq** [comment, marked as answer]
Not particularly a turbo specific problem, but you could create a root level config (or a package) for your base build config and extend that config in each package. (this is probably what I would do)

Alternatively you could create a separate `build-tools` package that builds a utility to [programmatically](https://rollupjs.org/guide/en/#javascript-api) run your build for the given package folder. So you could run something like this for each package:
```js
"scripts": {
 "build": "custom-build --<any-custom-options>"
}
```
With this approach turbo would require you to list this a `dev-dependency` for each package so it knows to build that package first before everything else.
